### ISSUE TITLE
`stack level too deep (SystemStackError)` after upgrading from 4.6.1 to 4.6.2

### ISSUE THREAD

**Issue by knapo** [getsentry/sentry-ruby#1520]
**Describe the bug**
Per https://github.com/getsentry/sentry-ruby/commit/4dc603b4f45eb2f34ed2b283fa3084360b067f32#commitcomment-54080150

`4.6.2` extends `Rake::Task` with aliasing `execute` method. Unfortunately this causes `stack level too deep (SystemStackError)`, when `Rake::Task#execute` is overridden using `#prepend`

**To Reproduce**

Create a `Rakefile`:

```
require 'bundler/inline'
require 'rake'

module MyGem
  module Task
    def execute(...)
      # do something
      super(...)
    end
  end
end

Rake::Task.prepend(MyGem::Task)

require 'sentry-ruby'

task(:test) { }
```

Run `rake test`

**Expected behavior**

`rake test` should not cause `SystemStackError`

**Actual behavior**

`rake test` ends with `SystemStackError`

**Environment**
 - Ruby Version: 3.0.2

**Comment by xiazek** [comment]
👍  Similar problem on my project.

**Comment by st0012** [comment]
Thanks for raising the issue to also let other users add reaction to it 👍 

I just checked a few similar gems that have `rake` integration. These are `prepend-only` ones:

- [Datadog](https://github.com/DataDog/dd-trace-rb/blob/b55af8ab7af8e54647245db2f268b569b9537b1d/lib/ddtrace/contrib/rake/instrumentation.rb#L9-L11)
- [Appsignal](https://github.com/appsignal/appsignal-ruby/blob/4940d3b87bdcff595674e5dcd9407f2337d8b107/lib/appsignal/hooks/rake.rb#L15)

`newrelic-ruby-agent` supports both method aliasing and prepend, but since `7.0` the default would be prepend.

And `scout_apm`, which uses method aliasing for patching classes (see #1427 for more detail), doesn't have rake integration.

So I think it's relatively safe to move from method aliasing to `prepend` 🙂

**Comment by st0012** [comment]
@knapo thanks again for the reporting and PR. I've released `4.6.4`, which contains the fix 🙂

**Comment by knapo** [comment]
Great, thx a lot too!
